### ISSUE TITLE
Change users domain permissions

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#194]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/194) by Matthew Lagoe (Bitbucket: [botanic](https://bitbucket.org/botanic), GitHub: [botanic](https://github.com/botanic)).**

----------------------------------------

This requires setting AccessPrivilege to Closed for all shards in the nel permissions table

only issue i have is how to get the shard list

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

What do you mean by getting the shards list?

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

The list of domains which users need to get access to when they register should be configurable from inside the AMS directly.

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Adjusting this access list should automatically reconfigure all user access as well (reconfigure all users who do not have :DEV: inside their shard permissions string).

**Comment by ryzom-pipeline** [comment]
**Original comment by Matthew Lagoe (Bitbucket: [botanic](https://bitbucket.org/botanic), GitHub: [botanic](https://github.com/botanic)).**

----------------------------------------

```
#!c++

<Kaetemi> and update the nel database properly when changing the macro permission in ams
<Botanic> just client app xyz
<Botanic> shardid -1
<Botanic> access priv OPEN
<Kaetemi> yeah
<Botanic> and set the uid
<Kaetemi> admins need access priv CLOSED or LOCKED or something so they can join when the shard is locked :)
<Kaetemi> and moderators
<Botanic> ooh 
<Botanic> makes sence
<Botanic> so should i set :DEV: on admin/mod?
<Botanic> as well
<Kaetemi> on admins
<Botanic> mod are basicallyCSR 
<Botanic> they get any special priv?
<Kaetemi> give them :GM: or something :)
<Botanic> ok admin :DEV: mod :GM:
<Kaetemi> yeah
<Kaetemi> that's simplest for now
<Botanic> and admin LOCKED mod CLOSED?
<Kaetemi> i don't know which of these values it is though..
<Kaetemi> one moment
<Botanic> like i do!
<Botanic> :P
<Botanic> i think ill make a shard list page
<Botanic> with default perms
<Botanic> that you can change
<Botanic> as well as a subpage that lists all users with permissions
<Kaetemi> values are
<Kaetemi> CLOSE, DEV, RESTRICTED, OPEN
<Kaetemi> admin -> DEV
<Kaetemi> mod -> RESTRICTED
<Kaetemi> user -> OPEN
<Kaetemi> basically AMS needs
<Kaetemi> (1) A configurable list of domains to which users get permissions on registration
<Kaetemi> which 
<Kaetemi> does
<Kaetemi> INSERT INTO `permission` (`UId`, `ClientApplication`) VALUES ('123', 'domain_name');
<Kaetemi> when the user is banned basically need to delete all the rows matching UID and ClientApplication for said domain names
<Kaetemi> unban, adds back in
<Kaetemi> when the configurable list of domains is changed need to add all permisions for all users which aren't banned :)
<Kaetemi> for moderators, this is INSERT INTO `nel`.`permission` (`UId`, `ClientApplication`, `AccessPrivilege`) VALUES ('123', 'domain_name', 'OPEN,RESTRICTED');
<Kaetemi> for devs OPEN,RESTRICTED,DEV
```

**Comment by ryzom-pipeline** [comment]
**Original comment by Matthew Lagoe (Bitbucket: [botanic](https://bitbucket.org/botanic), GitHub: [botanic](https://github.com/botanic)).**

----------------------------------------


```
#!c++

<Botanic>  INSERT INTO `nel`.`permission` (`UId`, `ClientApplication`, `AccessPrivilege`) VALUES ('123', 'domain_name', '');
<Botanic> what will that do?
<Kaetemi> then the user doesn't have permission :)
<Botanic> sure
<Botanic> so there banned!
<Kaetemi> make that banned then
<Botanic> ya
<Kaetemi> and what's the difference with no line?
<Botanic> :)
<Botanic> no permission
<Botanic> ie a private domain
<Botanic> like a test server
<Botanic> orw/e
<Kaetemi> okay
<Kaetemi> got it
<Kaetemi> so what if a user is banned on all my domains and i add a new domain
<Kaetemi> to the global access list
<Kaetemi> that is
<Botanic> well ill add a ams global banned flag
<Botanic> as well
<Kaetemi> ok
<Botanic> the '' priv is for per domain banned
<Botanic> not global
<Botanic> also doing it that way means i can add a "add permissions to all users" button for a domain
<Botanic> if you turn a domain into public
<Botanic> while not unbanning people
```

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

ClientApplication is now DomainId

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

The plugin gives

[07-Sep-2014 22:33:13 UTC] PHP Fatal error:  Allowed memory size of 134217728 bytes exhausted (tried to allocate 130968 bytes) in /srv/data/web/vhosts/*/private_php/ams/smarty/libs/sysplugins/smarty_internal_templatebase.php on line 286

Something seriously going wrong there?

**Comment by ryzom-pipeline** [comment]
**Original comment by Jan Boon (Bitbucket: [Jan Boon](https://bitbucket.org/Jan Boon), ).**

----------------------------------------

Fatal error: Uncaught exception 'SmartyCompilerException' with message 'Syntax Error in template "/srv/data/web/vhosts/*/private_php/ams/plugins/Domain_Management/templates/index.tpl" on line 203 "&lt;td&gt;&lt;a href=&quot;{$hook_info['Domain_Management']['path']}?page=layout_plugin&amp;name=Domain_Management&amp;edit_domain={$array['domain_id']}&quot;&gt;{$array['description']|truncate}&lt;/a&gt;&lt;/td&gt;" unknown modifier "truncate"' in /srv/data/web/vhosts/*/private_php/ams/smarty/libs/sysplugins/smarty_internal_templatecompilerbase.php:665 Stack trace: #0 /srv/data/web/vhosts/*/private_php/ams/smarty/libs/sysplugins/smarty_internal_compile_private_modifier.php(132): Smarty_Internal_TemplateCompilerBase->trigger_template_error('unknown modifie...', 203) #1 /srv/data/web/vhosts/*/private_php/ams/smarty/libs/sysplugins/smarty_internal_templatecompilerbase.php(48 in /srv/data/web/vhosts/*/private_php/ams/smarty/libs/sysplugins/smarty_internal_templatecompilerbase.php on line 665

**Comment by ryzom-pipeline** [comment]
**Original comment by Matthew Lagoe (Bitbucket: [botanic](https://bitbucket.org/botanic), GitHub: [botanic](https://github.com/botanic)).**

----------------------------------------

Fix #194
